src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,51 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidRegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int MIN_LOGIN_LENGTH = 6;
+    public static final int MIN_PASSWORD_LENGTH = 6;
+    public static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidRegistrationException("User can't be null");
+        }
+
+        if (user.getLogin() == null) {
+            throw new InvalidRegistrationException("User's login can't be null");
+        }
+
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new InvalidRegistrationException("User's login"
+                    + " can't have less than 6 character's");
```

Comment:
There is a grammatical error in the exception message: "character's" should be "characters". This is an issue related to the task requirements for clear and correct error messages.